File: basic/src/main/java/com/yuanshenbin/basic/state/TimeOutState.java
package com.yuanshenbin.basic.state;

import android.support.annotation.DrawableRes;

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 超时状态
 */

public abstract class TimeOutState extends StateAbstract {

    public abstract void showTimeOut(@DrawableRes int resId, String... msg);

    public abstract void showTimeOut(@DrawableRes int resId);

    public abstract void showTimeOut(String... msg);
}

File: basic-kotlin/src/main/java/com/yuanshenbin/basic/state/TimeOutState.kt
package com.yuanshenbin.basic.state

import androidx.annotation.DrawableRes

/**
 * author : yuanshenbin
 * time   : 2017/12/7
 * desc   : 超时状态
 */
abstract class TimeOutState : StateAbstract() {
    abstract fun showTimeOut(@DrawableRes resId: Int, vararg msg: String?)
    abstract fun showTimeOut(@DrawableRes resId: Int)
    abstract fun showTimeOut(vararg msg: String?)
}